### PR TITLE
RCDs now use right click to deconstruct

### PR DESCRIPTION
**Pull request by TJatPBnJ**
## About The Pull Request
RCDs use right click to deconstruct, everything else is functionally the same. The RCD defines also aren't bitflags anymore, they didn't seem to have any reason to be in the first place.

## Why It's Good For The Game
What's probably the most used mode of the RCD is now bound to the mouse, which is good.

## Changelog
:cl:
tweak: RCDs deconstruct with right click, instead of it being an option from the menu.
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
This is gonna save a bunch of time

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -766,9 +771,19 @@ RLD
...
 	if(!range_check(A,user))
 		return
-	if(target_check(A,user))
-		user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
-	rcd_create(A,user)
+	pre_attack(A, user)
+
+/obj/item/construction/rcd/arcd/afterattack_secondary(atom/A, mob/user)
+	if(!range_check(A,user))
+		return
```
Comment (Jared-Fogle) [inline code/game/objects/items/RCD.dm:778]: Return something here.

File: code/game/objects/items/RCD.dm [hunk 2]
```diff
@@ -766,9 +771,19 @@ RLD
...
+	pre_attack_secondary(A, user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+
+/obj/item/construction/rcd/arcd/rcd_create(atom/A, mob/user)
+	. = ..()
+	user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
+	return
```
Comment (Jared-Fogle) [inline code/game/objects/items/RCD.dm:786]: Remove this, trailing returns are pointless.

File: code/game/objects/items/RCD.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/game/objects/items/RCD.dm:778]: Probably SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN.

File: code/game/objects/items/RCD.dm [hunk 3]
```diff
@@ -766,9 +771,19 @@ RLD
 	. = ..()
 	if(!range_check(A,user))
 		return
-	if(target_check(A,user))
-		user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
-	rcd_create(A,user)
+	pre_attack(A, user)
+
+/obj/item/construction/rcd/arcd/afterattack_secondary(atom/A, mob/user)
```
Comment (Jared-Fogle) [inline code/game/objects/items/RCD.dm:776]:
```suggestion
/obj/item/construction/rcd/arcd/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
```

Do not rename the parameters, it breaks named parameters. Also, it's ugly.

File: code/game/objects/items/RCD.dm [hunk 4]
```diff
@@ -643,11 +641,18 @@ RLD
...
-/obj/item/construction/rcd/afterattack(atom/A, mob/user, proximity)
+/obj/item/construction/rcd/pre_attack(atom/A, mob/user, params)
 	. = ..()
-	if(!prox_check(proximity))
-		return
+	mode = constructionmode
 	rcd_create(A, user)
+	return FALSE
+
+/obj/item/construction/rcd/pre_attack_secondary(atom/A, mob/user, params)
```
Comment (Jared-Fogle) [inline code/game/objects/items/RCD.dm:650]:
```suggestion
/obj/item/construction/rcd/pre_attack_secondary(atom/target, mob/living/user, params)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Jared-Fogle** [comment]
you have to fix your code too i just suggested new function signatures 😄

**Comment by TJatPBnJ** [comment]
i realized

File: code/game/objects/items/RCD.dm (hunk 2 above)
Comment (TJatPBnJ) [reply code/game/objects/items/RCD.dm]: I'll have it return what the original proc returned, which is probably what I intended it to do.

File: code/game/objects/items/RCD.dm [hunk 5]
```diff
@@ -769,21 +769,20 @@ RLD
...
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 
 /obj/item/construction/rcd/arcd/rcd_create(atom/A, mob/user)
 	. = ..()
-	user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
-	return
+	if(.)
+		user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
+	return .
```
Comment (Jared-Fogle) [inline code/game/objects/items/RCD.dm:785]: Don't need this, it'll auto-return `.`

**Review by Mothblocks — APPROVED** [review]

File: code/game/objects/items/RCD.dm [hunk 6]
```diff
@@ -214,6 +214,7 @@ RLD
 	item_flags = NO_MAT_REDEMPTION | NOBLUDGEON
 	has_ammobar = TRUE
 	var/mode = RCD_FLOORWALL
+	var/constructionmode = RCD_FLOORWALL
```
Comment (Rohesie) [inline code/game/objects/items/RCD.dm:217]:
```suggestion
	var/construction_mode = RCD_FLOORWALL
```
`snake_case` is better then `nospacescase`

File: code/game/objects/items/RCD.dm [hunk 7]
```diff
@@ -580,17 +580,17 @@ RLD
...
 		"Furnishing" = image(icon = 'icons/hud/radial.dmi', icon_state = "chair")
 		)
-	if(mode == RCD_AIRLOCK)
+	if(constructionmode == RCD_AIRLOCK)
 		choices += list(
 		"Change Access" = image(icon = 'icons/hud/radial.dmi', icon_state = "access"),
 		"Change Airlock Type" = image(icon = 'icons/hud/radial.dmi', icon_state = "airlocktype")
 		)
-	else if(mode == RCD_WINDOWGRILLE)
+	else if(constructionmode == RCD_WINDOWGRILLE)
```
Comment (Rohesie) [inline code/game/objects/items/RCD.dm:588]: Can you turn this into a switch?

**Review by Rohesie — CHANGES_REQUESTED** [review]

**Comment by KathyRyals** [comment]
This PR might have broken some of the RCD's upgrades on live. Can't built machine frames and computer frames.

**Pull request merged**